api/api-notification-filter-v2/src/main/java/com/thoughtworks/go/apiv2/notificationfilter/NotificationFilterControllerV2.java

```diff
@@ -0,0 +1,202 @@
+
+    private final ApiAuthenticationHelper apiAuthenticationHelper;
+    private final UserService userService;
+    private final GoConfigService goConfigService;
+    private final EntityHashingService entityHashingService;
+
+    @Autowired
+    public NotificationFilterControllerV2(ApiAuthenticationHelper apiAuthenticationHelper,
+                                          UserService userService,
+                                          GoConfigService goConfigService,
+                                          EntityHashingService entityHashingService) {
+        super(ApiVersion.v2);
+        this.apiAuthenticationHelper = apiAuthenticationHelper;
+        this.userService = userService;
+        this.goConfigService = goConfigService;
+        this.entityHashingService = entityHashingService;
+    }
+
+    @Override
+    public String controllerBasePath() {
+        return Routes.NotificationFilterAPI.API_BASE;
+    }
+
+    @Override
+    public void setupRoutes() {
+        path(controllerBasePath(), () -> {
+            before("", mimeType, this::setContentType);
+            before("", mimeType, this::verifyContentType);
+            before("", mimeType, this.apiAuthenticationHelper::checkUserAnd403);
+            before("", mimeType, this::canConfigureNotificationFilters);
+
+            before("/*", mimeType, this::verifyContentType);
+            before("/*", mimeType, this::setContentType);
+            before("/*", mimeType, this.apiAuthenticationHelper::checkUserAnd403);
+            before("/*", mimeType, this::canConfigureNotificationFilters);
+
+            get("", mimeType, this::index);
+            get(Routes.NotificationFilterAPI.ID, mimeType, this::show);
+            post("", mimeType, this::createNotificationFilter);
+            patch(Routes.NotificationFilterAPI.ID, mimeType, this::updateFilter);
+            delete(Routes.NotificationFilterAPI.ID, mimeType, this::deleteFilter);
+        });
+    }
+
+    public String index(Request request, Response response) throws IOException {
+        User user = userService.findUserByName(currentUsernameString());
+        List<NotificationFilter> notificationFilters = user.getNotificationFilters();
+
+        return writerForTopLevelObject(request, response, writer -> NotificationFiltersRepresenter.toJSON(writer, notificationFilters));
+    }
+
+    public String show(Request request, Response response) throws IOException {
+        long id = getIdFromRequest(request.params("id"));
+        User user = userService.findUserByName(currentUsernameString());
+        Optional<NotificationFilter> optionalFilter = user.getNotificationFilters().stream()
+                .filter(filter -> filter.getId() == id)
+                .findFirst();
+
+        if (optionalFilter.isEmpty()) {
+            throw new RecordNotFoundException(EntityType.NotificationFilter, id);
+        }
```

Comment:
The logic is somewhat similar to `doFetchEntityFromConfig`. Maybe, replace it with that?